fsspec/implementations/reference.py

```diff
@@ -199,9 +200,12 @@ async def _get_file(self, rpath, lpath, **kwargs):
             f.write(data)
 
     def get_file(self, rpath, lpath, **kwargs):
+        callback = as_callback(kwargs.pop("callback", None))
         data = self.cat_file(rpath, **kwargs)
```

Comment:
Right, though there is no guarantee any of the functions will have the true 'progress'. I didn't try to spend too much on the external implementations like reference, but can certainly try to make it improved.

Reply:
It seems like it is not easily possible to show true progress, since `get_file` needs to be about both reading+writing, I don't think we should show progress of cat_file. So I left it as is.

Reply:
I think w are safe to assume that the reading will never be faster than the writing, and usually much slower, so maybe showing the progress of cat_file is the right thing to do after all.

Reply:
Btw i just noticed that `ReferenceFileSystem.cat_file` doesn't show progress, since it is just `f.read()`. So I am not sure how we can achieve that here.